code/modules/admin/tickets/admin_ticket.dm

```diff
@@ -0,0 +1,151 @@
+
+var/list/tickets_list = list()
+var/ticket_count = 0
+var/ticket_counter_visible_to_everyone = 0
+
+/datum/ticket_log
+	var/datum/admin_ticket/parent
+	var/gametime
+	var/user
+	var/user_admin = 0
+	var/text
+	var/text_admin
+	var/for_admins
+
+/datum/ticket_log/New(var/datum/admin_ticket/parent, var/user, var/text, var/for_admins = 0)
+	src.gametime = gameTimestamp()
+	src.parent = parent
+	if(istype(user, /client))
+		src.user_admin = is_admin(user)
+	else
+		src.user_admin = text
```

Comment:
Based on this line, it's a fallback should the istype fail above, such as the client suddenly going missing midprocess. Should be fine.